apps/game_backend/lib/game_backend/users.ex

```diff
@@ -168,24 +168,19 @@ defmodule GameBackend.Users do
 
   defp quests_preloads(base_query) do
     naive_today = NaiveDateTime.utc_now()
-    start_of_date = NaiveDateTime.beginning_of_day(naive_today)
-    end_of_date = NaiveDateTime.end_of_day(naive_today)
-
     start_of_week = Date.beginning_of_week(NaiveDateTime.to_date(naive_today), :sunday)
-    end_of_week = Date.add(start_of_week, 6)
     {:ok, start_of_week_naive} = NaiveDateTime.new(start_of_week, ~T[00:00:00])
-    {:ok, end_of_week_naive} = NaiveDateTime.new(end_of_week, ~T[23:59:59])
+    end_of_week_naive = NaiveDateTime.add(start_of_week_naive, 7, :day)
 
     quests_subquery =
       from(user_quest in UserQuest,
         as: :user_quest,
         join: quest in assoc(user_quest, :quest),
         as: :quest,
         where:
-          (quest.type in ^["daily", "milestone"] and user_quest.inserted_at > ^start_of_date and
```

Comment:
This was the main culprit for the `0/6` error, it was preventing the fetching of previous day quests so as a result it was always showing as 0